controllers/addon/addon_controller.go

```diff
@@ -402,3 +402,57 @@ func (r *AddonReconciler) reconcile(ctx context.Context, addon *addonsv1alpha1.A
 
 	return ctrl.Result{}, nil
 }
+
+// Lists and filters pods with corev1.PodReasonUnschedulable status
+func (r *AddonReconciler) listUnschedulableAddonPods(
+	ctx context.Context,
+	addon *addonsv1alpha1.Addon,
+) (*corev1.PodList, error) {
+	addonPods := &corev1.PodList{}
+	unschedulablePods := &corev1.PodList{}
+
+	if addon.Status.Phase == addonsv1alpha1.PhaseReady {
+		return unschedulablePods, nil
+	}
+
+	if err := r.Client.List(
+		ctx,
+		addonPods,
+		client.InNamespace(addon.Spec.Install.OLMOwnNamespace.Namespace),
+	); err != nil {
+		return unschedulablePods, fmt.Errorf("failed listing addon pods: %w", err)
+	}
+
+	for _, pod := range addonPods.Items {
```

Comment:
How about we check for kube events in the addon namespace instead of iterating through the pod metadata? This way we can have a simpler approach and concise code.

Reply:
@supreeth7 Can you explain why checking for events is simpler? Maybe I don't know what you mean. 

If we use events, we list and loop through them, and ensure each of the following;

1. an event is a failedscheduling event
2. an event references at least an addon's pod
3. an event is "recent"

Is there a way to do this in a "simple approach"?

Reply:
We can list filtered events making it more or less a one-liner; we should narrow it down to fetch the most recent event [0] that indicates that a pod is still pending.

```
events := &corev1.EventList{}

r.Client.List(ctx, events, &client.ListOptions{
		Namespace: addon.Spec.Install.OLMOwnNamespace.Namespace,
		LabelSelector: "ado-label"
	)
```
You can play with `LabelSelector` or `FieldSelector` to find the right filter.

Reply:
@supreeth7 I don't disagree with a one liner, however I prefer a reliable and practical method.

Here's my explanation why. A k8s event is just a side-effect, it should never be actionable - it's not a state. On the other hand, if an event is used, it should be referenced back to the object that it has referenced to - this is a best practice. Event is similar to a log, they aren't states but indicators only, hence aren't completely reliable.

The one liner you've mentioned above I've already tried it (it won't be a one-liner) before I created this PR, I ended up referencing it back to one of the addon's pods which is looping it. Secondly an event is temporal you ahve the following sequence;

event1
event2
failedscheduling1
event3
event4

At event4, the pod may already be schedulable but listing the eevents will give you failedscheduling1 and using this as a reason is not correct. 

That is looping at the pods and checking their conditions is a reliable approach because that is a state that is asserted by k8s, not just a side-effect.

Reply:
I agree with @pepedocs here as kubernetes events have low retention which can lead to flapping or inconsistent state issues. The pod controller however should maintain relatively stable status conditions in contrast.